sql-plugin/src/main/scala/com/nvidia/spark/rapids/HostAlloc.scala

```diff
@@ -415,7 +416,9 @@ object HostAlloc {
   }
 
   def alloc(amount: Long, preferPinned: Boolean = true): HostMemoryBuffer = {
-    getSingleton.alloc(amount, preferPinned)
+    GpuTaskMetrics.get.hostAllocTime {
```

Comment:
Do we really want this metric?  If we're not spilling, this should be immeasurably small.  If we are spilling, we already have metrics to cover time spent spilling.

Reply:
@revans2 that was my interpretation of

> We will have one metric for the amount of time a task was blocked on host memory allocation;

in #8880. Did I misinterpret it?